### PR TITLE
Profile/aw/62940/add authenticated menu links

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- We are adding some more links to the authenticated menu.
- This work lives behind a feature toggle: `show_authenticated_menu_enhancements`
- 2 list items added to the menu for Dependents and Letters
- Borders added to top and bottom of some items to separate grouped items. 

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
https://github.com/department-of-veterans-affairs/va.gov-team/issues/62940
- _Link to epic if not included in ticket_
[#621](https://github.com/department-of-veterans-affairs/va.gov-team/issues/62186)

## Testing done

- Added unit test for added links

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2023-08-25 at 1 29 00 PM | Screenshot 2023-08-25 at 1 28 39 PM |
| Desktop | Screenshot 2023-08-25 at 1 29 13 PM | Screenshot 2023-08-25 at 1 27 06 PM |

## What areas of the site does it impact?

Header of site

## Acceptance criteria

- [ ] Additional links are shown when toggle is on
- [ ] Menu remains unchanged on production at this time, since this will be tested in staging first.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [x] The vets-website header does not contain any web-components
- [x] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [x] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

I couldn't seem to run the testing steps for the proxy-rewrite for header and footer. Are there new steps to getting the tests to run? Also is there any issue with using a feature toggle within the header?

### CODE REVIEW

**Review by powellkerry — APPROVED** [review]
Everything looks good once the PropTypes.node duplicate is removed.

File: src/platform/site-wide/user-nav/components/PersonalizationDropDownPanel.jsx [hunk 1]
```diff
@@ -67,7 +68,11 @@ PersonalizationDropDownPanel.propTypes = {
   /**
    * The text of the drop down button.
    */
-  buttonText: PropTypes.string.isRequired,
+  buttonText: PropTypes.oneOfType([
+    PropTypes.node,
+    PropTypes.string,
+    PropTypes.node,
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/platform/site-wide/user-nav/components/PersonalizationDropDownPanel.jsx:74]: whoops! will correct, thanks

**Review by randimays — APPROVED** [review]
LGTM

**Review by allisonlu — APPROVED** [review]
🚀 lgtm

**Pull request merged**